iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TerminalRefXml.java

```diff
@@ -18,21 +19,34 @@
  */
 public final class TerminalRefXml {
 
-    public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String elementName) throws XMLStreamException {
-        writeTerminalRef(t, context, context.getVersion().getNamespaceURI(), elementName);
+    public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String elementName, String id) throws XMLStreamException {
+        writeTerminalRef(t, context, context.getVersion().getNamespaceURI(), elementName, id);
     }
 
-    public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String namespace, String elementName) throws XMLStreamException {
-        writeTerminalRef(t, context, namespace, elementName, context.getWriter());
+    public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String namespace, String elementName, String id) throws XMLStreamException {
+        writeTerminalRef(t, context, namespace, elementName, id, context.getWriter());
     }
 
     public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String namespace, String elementName, XMLStreamWriter writer) throws XMLStreamException {
+        writeTerminalRef(t, context, namespace, elementName, null, writer);
+    }
+
+    public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String namespace, String elementName, String id, XMLStreamWriter writer) throws XMLStreamException {
         Connectable c = t.getConnectable();
         if (!context.getFilter().test(c)) {
             throw new PowsyblException("Oups, terminal ref point to a filtered equipment " + c.getId());
         }
         writer.writeEmptyElement(namespace, elementName);
-        writer.writeAttribute("id", context.getAnonymizer().anonymizeString(c.getId()));
+        final boolean[] writeId = {false};
+        IidmXmlUtil.runUntilMaximumVersion(IidmXmlVersion.V_1_2, context, () -> writeId[0] = true);
+        IidmXmlUtil.runFromMinimumVersion(IidmXmlVersion.V_1_3, context, () -> {
+            if (id == null || !id.equals(c.getId())) {
+                writeId[0] = true;
+            }
+        });
+        if (writeId[0]) {
+            writer.writeAttribute("id", context.getAnonymizer().anonymizeString(c.getId()));
+        }
```

Comment:
This code looks complicated for nothing:
```
IidmXmlUtil.runUntilMaximumVersion(IidmXmlVersion.V_1_2, context, () -> {
    writer.writeAttribute("id", context.getAnonymizer().anonymizeString(c.getId()));
}
IidmXmlUtil.runFromMinimumVersion(IidmXmlVersion.V_1_3, context, () -> {
    if (id == null || !id.equals(c.getId())) {
        writer.writeAttribute("id", context.getAnonymizer().anonymizeString(c.getId()));
    }
});
```

Reply:
I made this to avoid 2 try-catch

Reply:
@MioRtia What do you think to create utility functions to write attributes that throws only unchecked exception. Or to introduce a new utility function like the one you created for integer  in IidmXmlUtil (writeIntAttributeUntilMaximumVersion)

Reply:
@jlabous We propose to add a writeStringAttribute in XmlUtil class that catch XmlStreamException and wrap them into UncheckedXmlStreamException, and to use this method in the lambdas.
Is it ok for you?

Reply:
Ok I'll go on this